shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java

```diff
@@ -61,31 +66,62 @@ public void close() {}
   @Override
   public InterpreterResult interpret(String cmd, InterpreterContext contextInterpreter) {
     logger.debug("Run shell command '" + cmd + "'");
-    long start = System.currentTimeMillis();
     CommandLine cmdLine = CommandLine.parse("bash");
     cmdLine.addArgument("-c", false);
     cmdLine.addArgument(cmd, false);
     DefaultExecutor executor = new DefaultExecutor();
     ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    executor.setStreamHandler(new PumpStreamHandler(outputStream));
-
+    ByteArrayOutputStream errorStream = new ByteArrayOutputStream();
+    executor.setStreamHandler(new PumpStreamHandler(outputStream, errorStream));
     executor.setWatchdog(new ExecuteWatchdog(commandTimeOut));
+
+    Job runningJob = getRunningJob(contextInterpreter.getParagraphId());
+    Map<String, Object> info = runningJob.info();
+    info.put(EXECUTOR_KEY, executor);
     try {
-      int exitValue = executor.execute(cmdLine);
+      executor.execute(cmdLine);
```

Comment:
maybe just log it for now, it would be nice if this could be an interpreter variable that could be referred back